src/components/NewGalaxyModal.js

```diff
@@ -96,32 +145,68 @@ export const NewGalaxyModal = _.flow(
-        span({ style: { fontWeight: 600 } }, ['delete any files on the associated hard disk']),
-        ' (e.g. results files). Double check that your workflow results were written to ',
-        'the data tab in the workspace before clicking “Delete”'
-      ]),
-      div({ style: { marginTop: '1rem' } }, [
-        'Deleting your Cloud Environment will stop your ',
-        'running Galaxy application and your application costs. You can create a new Cloud Environment ',
-        'for Galaxy later, which will take 8-10 minutes.'
+    return h(Fragment, [
+      h(TitleBar, {
+        title: 'Delete Cloud Environment for Galaxy',
+        style: { marginBottom: '0.5rem' },
+        onDismiss,
+        onPrevious: !!viewMode ? () => setViewMode(undefined) : undefined
+      }),
+      div({ style: { lineHeight: '22px' } }, [
+        div({ style: { marginTop: '0.5rem' } }, [
+          'Deleting your Cloud Environment will also ',
+          span({ style: { fontWeight: 600 } }, ['delete any files on the associated hard disk']),
+          ' (e.g. results files). Double check that your workflow results were written to ',
+          'the data tab in the workspace before clicking “Delete”'
+        ]),
+        div({ style: { marginTop: '1rem' } }, [
+          'Deleting your Cloud Environment will stop your ',
+          'running Galaxy application and your application costs. You can create a new Cloud Environment ',
+          'for Galaxy later, which will take 8-10 minutes.'
+        ])
       ])
     ])
   }
 
   const renderLaunchWarning = () => {
-    return div({ style: { lineHeight: '22px' } }, [
-      h(GalaxyWarning)
+    return h(Fragment, [
+      h(TitleBar, {
+        title: h(WarningTitle, ['Launch Galaxy']),
+        style: { marginBottom: '0.5rem' },
+        onDismiss,
+        onPrevious: !!viewMode ? () => setViewMode(undefined) : undefined
+      }),
+      div({ style: { lineHeight: '22px' } }, [
+        h(GalaxyWarning)
+      ])
     ])
   }
 
+  const getEnvMessageBasedOnStatus = isTitle => {
+    return Utils.cond(
+      [!!app && (app.status === 'STOPPED'), () => `Cloud environment is now paused ${(!isTitle ? '...' : '')}`],
+      [!!app && (app.status === 'PRESTOPPING'), () => 'Cloud environment is preparing to stop.'],
+      [!!app && (app.status === 'STOPPING'), () => `Cloud environment is pausing. ${!isTitle ? 'This process will take up to a few minutes' : ''}`],
+      [!!app && (app.status === 'PRESTARTING'), () => 'Cloud environment is preparing to start.'],
+      [!!app && (app.status === 'STARTING'), () => `Cloud environment is starting. ${!isTitle ? 'This process will take up to a few minutes' : ''}`],
+      [!!app && (app.status === 'ERROR'), () => `An error has occurred on your Cloud Environment. ${!isTitle ? 'This process will take up to a few minutes' : ''}`],
+      () => isTitle ? 'Cloud environment' : `Environment ${app ? 'consists' : 'will consist'} of an application and cloud compute.`
```

Comment:
Two comments:
1. instead of `!!app && app.foo`, you can use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining): `app?.foo`
2. This should be a `Utils.switchCase`; you can wrap it in a ternary checking if `app` is truthy.

Reply:
A question here about syntax:

For 1)

Would this look like: app?.status === "PRESTOPPING"

For 2)
Would we want the switchCase to depend on the app.status in this case?

Reply:
I was able to get the optional chaining to work, but having some trouble changing the block into a switchCase based on app?.status